mobile/src/main/java/org/openhab/habdroid/core/connection/AbstractConnection.java

```diff
@@ -84,14 +86,38 @@ public boolean checkReachabilityInBackground() {
             } else if (url.getProtocol().equals("https") && checkPort == -1) {
                 checkPort = 443;
             }
-            Socket s = new Socket();
-            s.connect(new InetSocketAddress(url.getHost(), checkPort), 1000);
+            Socket s = createConnectedSocket(new InetSocketAddress(url.getHost(), checkPort));
+            if (s == null) {
+                return false;
+            }
             Log.d(TAG, "Socket connected");
             s.close();
             return true;
         } catch (Exception e) {
-            Log.e(TAG, e.getMessage());
+            Log.d(TAG, e.getMessage());
             return false;
         }
     }
+
+    private Socket createConnectedSocket(InetSocketAddress socketAddress) {
+        Socket s = new Socket();
+        for (int retries = 0; retries < 10; retries++) {
+            try {
+                s.connect(socketAddress, 1000);
+            } catch (SocketTimeoutException e) {
+                Log.d(TAG, "Socket timeout at the " + retries + ". try.", e);
```

Comment:
`5. try` is not valid English, as `5.` is a German construct. Could write `Socket timeout after 5 retries` here.

Also, I'm not sure whether printing the stack trace here (and below) is actually useful.

Reply:
> Also, I'm not sure whether printing the stack trace here (and below) is actually useful.

No, I don't think so.